### ISSUE TITLE
Code coverage of R itself

### ISSUE THREAD

**Issue by HenrikBengtsson** [RConsortium/wishlist#9]
Doing code coverage analysis on [the R source code](https://github.com/wch/r-source/commits/trunk) would:

* help identifying core code that is poorly tested or not tested at all
* be a great guide for writing additional tests
* these new tests will result in new bugs being identified
* new tests will lower the risk for new bugs being introduced
* ...
 
The [covr](https://cran.r-project.org/package=covr) package should be useful for this.  It is not obvious to me how to run covr on the core R packages since they're not standalone packages per se, e.g.
```r
> c <- covr::package_coverage(quiet = FALSE)
Error: '[...]/r-source/src/library/stats' does not contain a package!
```

**Comment by jimhester** [comment]
I agree this could be very useful, e.g. [rho](https://codecov.io/github/rho-devel/rho?branch=master) tracks C/C++ coverage for instance.

Tracking the R code with covr is a little tricky though, many of the tests in base R are not in the individual packages themselves, but in the https://github.com/wch/r-source/tree/trunk/tests directory, the packages (particuarly the base package) uses different methods to load than a user package and covr obviously uses base functions when it runs, which may muddle the counts.
